asar-teensy-firmware/TestScripts/XBEE/Xbee_read/Xbee_read.ino

```diff
@@ -0,0 +1,161 @@
+* It gets called every sPeriod Seconds                                                */
+void callback()                                     
+{
+  digitalWrite(LED, digitalRead(13) ^ 1); //toggle LED
+  Flag = true;                            //Raise Flag
+}
+
+/*Read a single Instruction from xbee*/
+void readRawInstruct()
+{
+  if (XBEE.available() && XBEE.read() == 0xFF)               //Only if there is something from xbee to read
+  {
+    for (int i = 0; i <= 5; i++)        //Read all the inputs (until stop bit is read)
+    {
+      rawRead[i] = XBEE.read();             //and fill up the temporary array
+      //Serial.print("Argument # "); Serial.print(i);
+      //Serial.print(", Raw reading: "); Serial.println(rawRead[i], HEX); 
+      if(rawRead[i] == 0xF0)
+      {
+        i = 6;
+      }
+      else if (i >= 4)
+      {
+        Serial.println("ERROR-001: Stop Bit Not Detected");
+      }
+    }
+  }                                   
+}
+
+/*Interprets a single raw reading instruction, fills relevant arrays 
+ * with the index being the number of instruction */
+void Interpret_instruct()
+{
+  action[instructNum] = rawRead[0]; //add the action to instruction set
+  switch (action[instructNum])      //switch between the different types of moves
+  {
+    case 0xAA : //Turn Left
+      angle[instructNum] = rawRead[1]; //add the relative turning angle to instruction set
+      if (rawRead[2] != 0xF0)          //If the stop bit isnt next, something went wrong
+      {
+        Serial.println("ERROR-002: Stop Bit Not Detected as expected"); //print error message
+      }
+      break;
+    case 0xBB : //Turn Right
+      angle[instructNum] = rawRead[1]; //add the turning angle to instruction set
+      if (rawRead[2] != 0xF0)          //If the stop bit isnt next, something went wrong     
+      {
+        Serial.println("ERROR-002: Stop Bit Not Detected as expected"); //print error message
+      }
+      break;            
+    case 0xCC : //Go Forward
+      distance[instructNum] = rawRead[1]; //Add the distance of move to instruction set
+      speedy[instructNum] = rawRead[2];   //add the speed of move to instruction set
+      if (rawRead[3] != 0xF0)             //If the stop bit isnt next, something went wrong
+      {
+        Serial.println("ERROR-002: Stop Bit Not Detected as expected"); //print error message
+      }
+      break;           
+    default: //If the action type is not recognized
+      Serial.println("ERROR-003: Move-type not not recognized"); //print error message
+      break;  
+  }
+  for (int i = 0; i < 5; i++) 
+  {
+    rawRead[i] = 0;
+  }
+}
+
+
+/* Function that prints full instruction set */
+void printInstructSet()
+{
+  for (int i = 1; i <= instructNum; i++) //For each instruction
+  {
+    Serial.print("Instruction: "); Serial.println(i);
+    switch (action[i])      //switch between the different types of moves
+    {
+      case 0xAA : //Turn Left
```

Comment:
Magic numbers.

Reply:
enum thing

Reply:
```
enum class ACTION
{
    TURN_LEFT = 0xAA,
    TURN_RIGHT = 0xBB,
    // and so on...
};
```

Is what I would recommend. To reference the raw values use `ACTION::TURN_LEFT`.